wire-ios-request-strategy/Sources/Synchronization/Decoding/EventPayloadDecoder.swift

```diff
@@ -0,0 +1,56 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+struct EventPayloadDecoder {
+
+    private let decoder: JSONDecoder
+
+    init(decoder: JSONDecoder = .defaultDecoder) {
+        self.decoder = decoder
+    }
+
+    func decode<T>(
+        _ type: T.Type,
+        from eventPayload: [AnyHashable: Any]
+    ) throws -> T where T: Decodable {
+        let data: Data
+
+        do {
+            data = try JSONSerialization.data(withJSONObject: eventPayload, options: [])
```

Comment:
To me it looks like to be a bit redundant.
The Codable implementation is said to use JSONSerialization under the hood so we are repeating a step here. But it's probably convenient, I assume.

Btw, do we need this in the app code or could it be moved to the test target?